var/spack/repos/builtin/packages/mathematica/package.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+import os
+
+
+class Mathematica(Package):
+    """Mathematica: high-powered computation with thousands of Wolfram Language
+       functions, natural language input, real-world data, mobile support.
+
+       Note: A manual download is required for Mathematica.
+       Spack will search your current directory for the download file.
+       Alternatively, add this file to a mirror so that Spack can find it.
+       For instructions on how to set up a mirror, see
+       http://spack.readthedocs.io/en/latest/mirrors.html"""
+
+    homepage = "https://www.wolfram.com/mathematica/"
+    url = 'file:///{0}/Mathematica_12.0.0_LINUX.sh'.format(os.getcwd())
+
+    version('12.0.0', sha256='b9fb71e1afcc1d72c200196ffa434512d208fa2920e207878433f504e58ae9d7',
+            expand=False)
+
+    def install(self, spec, prefix):
+        sh = which('sh')
+        sh(self.stage.archive_file, '--', '-auto', '-verbose',
+           '-targetdir={0}'.format(prefix),
+           '-execdir={0}'.format(prefix.bin),
+           '-selinux=y')
+
+        # after the install phase completes, spack tries to install build.out
+        # into <prefix>/.spack, but the .spack dir will not exist, causing the
+        # build to fail. package.py:1690 seems to show that the dir is created
+        # right before writing build.out -- possible bug?
+
+        # creating the .spack dir right after installing prevents explosions
+        mkdirp(join_path(prefix, '.spack'))
```

Comment:
I have never encountered this bug before. Can you post the error message? @tgamblin @alalazo @scheibelp may be interested in this.

Reply:
Here is the output without the workaround after the installation script finishes (run with `spack -d`):
```
Installation complete.

<< Removing .InstallTemp-130069. >>
<< Exiting... >>
==> [2019-05-18-14:11:40.855543] Installing /home/cork/spack/var/spack/stage/mathematica-12.0.0-bc7l7wroj6jppgifoci6v2sypbcawye2/spack-build.out to /home/cork/spack/opt/spack/linux-rhel7-x86_64/gcc-4.8.5/mathematica-12.0.0-bc7l7wroj6jppgifoci6v2sypbcawye2/.spack/build.out
==> [2019-05-18-14:11:40.866659] WRITE LOCK: /home/cork/spack/opt/spack/.spack-db/prefix_lock[315075970839595387:1] [Released]
==> [2019-05-18-14:11:40.867008] WRITE LOCK: /home/cork/spack/var/spack/stage/.lock[1800670006513711093:1] [Released]
==> [2019-05-18-14:11:40.913805] Error: IOError: [Errno 2] No such file or directory: '/home/cork/spack/opt/spack/linux-rhel7-x86_64/gcc-4.8.5/mathematica-12.0.0-bc7l7wroj6jppgifoci6v2sypbcawye2/.spack/build.out'

/home/cork/spack/lib/spack/spack/package.py:1703, in log:
       1700            # FIXME : this potentially catches too many things...
       1701            pass
       1702
  >>   1703        # Archive the whole stdout + stderr for the package
       1704        install(self.log_path, log_install_path)
       1705        # Archive the environment used for the build
       1706        install(self.env_path, env_install_path)

See build log for details:
  /home/cork/spack/var/spack/stage/mathematica-12.0.0-bc7l7wroj6jppgifoci6v2sypbcawye2/spack-build.out
Traceback (most recent call last):
  File "/home/cork/spack/lib/spack/spack/build_environment.py", line 768, in child_process
    return_value = function()
  File "/home/cork/spack/lib/spack/spack/package.py", line 1542, in build_process
    self.log()
  File "/home/cork/spack/lib/spack/spack/package.py", line 1704, in log
    install(self.log_path, log_install_path)
  File "/home/cork/spack/lib/spack/llnl/util/filesystem.py", line 307, in install
    copy(src, dest, _permissions=True)
  File "/home/cork/spack/lib/spack/llnl/util/filesystem.py", line 290, in copy
    shutil.copy(src, dest)
  File "/usr/lib64/python2.7/shutil.py", line 119, in copy
    copyfile(src, dst)
  File "/usr/lib64/python2.7/shutil.py", line 83, in copyfile
    with open(dst, 'wb') as fdst:
IOError: [Errno 2] No such file or directory: '/home/cork/spack/opt/spack/linux-rhel7-x86_64/gcc-4.8.5/mathematica-12.0.0-bc7l7wroj6jppgifoci6v2sypbcawye2/.spack/build.out'
```

Reply:
Ping @scheibelp. Can you take a look at this? This is the only question holding up this PR as far as I'm concerned.